### PR TITLE
Add Developers Certificate of Origin

### PR DESCRIPTION
**Pull request by matthiasbeyer**
I'm not sure here: We do not have a license-header in the files itself, but a license file which lives in the repository.

We might have to fix this.

@neithernut , @TheNeikos and maybe @llogiq ... what do you think on the subject of having a license statement in all files - is it necessary?

### CODE REVIEW

**Comment by llogiq** [comment]
First the obligatory disclaimer: I'm not a lawyer, so I'm not qualified to speak on license issues. For what it's worth, I usually don't bother, and I follow project licenses when I learn of them. However, having a license note in each source file reduces the risk that one source file gets copied without the license header "by mistake", so some projects opt to put one in each source file. I don't mind either way.

**Comment by matthiasbeyer** [comment]
I guess I will add license headers in the files and ensure via CI that the header is present. Better safe than sorry!

**Comment by matthiasbeyer** [comment]
@TheNeikos do you see what's going on here? If you `git ls-files` in this branch, there is no "imag-diary", but travis reports it... I don't see why...

**Comment by matthiasbeyer** [comment]
Seems like the problem is that travis merges. Latest fixup tries to fix this by checking out the `$TRAVIS_COMMIT` after a successful `cargo build` step.

File: .travis.yml [hunk 1]
```diff
@@ -77,6 +76,9 @@ addons:
 after_success:
     - travis-cargo --only stable doc-upload
 
+after_script:
+    - source .lgpl_check.sh
```
Comment (matthiasbeyer) [inline .travis.yml]: @TheNeikos Will this actually break my build if the script calls `exit(1)`? I mean... does `after_script:` still cause a build failure if it fails on travis? I don't know...

Comment (TheNeikos) [inline .travis.yml]: Afaik it does not

Comment (matthiasbeyer) [inline .travis.yml]: @TheNeikos it seems like it actually does. Look at the latest evaluation. Or do you see something which failed the build besides the check?

**Comment by matthiasbeyer** [comment]
Okay, now we have the confirmation that it actually is not what we wanted. :cry: 

I guess I have to rewrite the testing script :cry:

**Comment by matthiasbeyer** [comment]
I guess this PR needs to be splitted. One for the Developers Certificate of Origin and the other for the license header.

**Comment by matthiasbeyer** [comment]
I moved the LGPL-Check stuff to #479, this PR is for the Developers Certificate of Origin only.

**Pull request merged**